src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -343,11 +343,15 @@ internal static Diagnostic Create(DiagnosticInfo info)
                 return null;
             }
 
+            if (compilation.SemanticModelProvider == null)
+            {
+                compilation = compilation.WithSemanticModelProvider(new CachingSemanticModelProvider());
```

Comment:
>compilation [](start = 16, length = 11)

It doesn't feel right to use a different compilation in SuppressMessageAttributeState